spec/std/http/server/handlers/static_file_handler_spec.cr

```diff
@@ -41,13 +48,55 @@ describe HTTP::StaticFileHandler do
     it "should serve file if file mtime is younger" do
       headers = HTTP::Headers.new
       headers["If-Modified-Since"] = HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time - 1.hour)
-      response = handle HTTP::Request.new("GET", "/test.txt")
+      response = handle HTTP::Request.new("GET", "/test.txt", headers)
       response.status_code.should eq(200)
       response.headers["Last-Modified"].should eq(HTTP.rfc1123_date(File.info("#{__DIR__}/static/test.txt").modification_time))
       response.body.should eq(File.read("#{__DIR__}/static/test.txt"))
     end
   end
 
+  context "with If-None-Match header" do
+    it "should return 304 Not Modified if header matches etag" do
+      etag = %{W/"#{File.info("#{__DIR__}/static/test.txt").modification_time.epoch.to_s}"}
+      headers = HTTP::Headers.new
+      headers["If-None-Match"] = etag
+      response = handle HTTP::Request.new("GET", "/test.txt", headers), ignore_body: true
+      response.headers["Etag"].should eq(etag)
+      response.status_code.should eq(304)
+    end
+
+    it "should serve file if header does not match etag" do
+      headers = HTTP::Headers.new
+      headers["If-None-Match"] = "some random etag"
+      response = handle HTTP::Request.new("GET", "/test.txt", headers)
+      response.status_code.should eq(200)
+      response.headers["Etag"].should match(/W\/"\d+"$/)
+      response.body.should eq(File.read("#{__DIR__}/static/test.txt"))
+    end
+  end
+
+  context "with both If-None-Match and If-Modified-Since headers" do
+    it "ignores If-Modified-Since as specified in RFC 7232" do
```

Comment:
How does this spec test if `If-Modified-Since` is ignored? The date needs to be before modifcation time of the file for this.

Reply:
Because of current 1s padding that I did not touched I don't think it's correct, see spec

> it "should return 304 Not Modified if file mtime is equal" do

Reply:
Oh, it seems there is something off... We need more integration tests for this.

```cr
ref_response = handle HTTP::Request.new("GET", "/test.txt")
headers = HTTP::Headers.new
headers["If-Modified-Since"] = ref_response.headers["Last-Modified"]
response = handle HTTP::Request.new("GET", "/test.txt")
```

This should result in 304 Not Modified (both with and without this PR), but it doesn't.